tools/lister.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import os
+import sys
+import subprocess
+import re
+from collections import defaultdict
+
+def get_ftype(fpath, use_shebang):
+    ext = os.path.splitext(fpath)[1]
+    if ext:
+        return ext[1:]
+    elif use_shebang:
+        # opening a file may throw an OSError
+        with open(fpath) as f:
+            first_line = f.readline()
+            if re.search(r'^#!.*\bpython', first_line):
+                return 'py'
+            elif re.search(r'^#!.*sh', first_line):
+                return 'sh'
+            elif re.search(r'^#!.*\bperl', first_line):
+                return 'pl'
+            elif re.search(r'^#!', first_line):
+                print('Error: Unknown shebang in file "%s":\n%s' % (fpath, first_line), file=sys.stderr)
+                return ''
+            else:
+                return ''
+    else:
+        return ''
+
+def list_files(targets=None, ftypes=None, use_shebang=True, modified_only=False,
+    exclude=None, group_by_ftype=False):
```

Comment:
This should be intended to start after the paren (see any other file with multi-line def for what I mean).